feat(actors_modal): Added breakdowns and series selectors to the actors modal

Gilbert09 opened the pull request:
## Problem
- We want to add a series, breakdown, date, and compare selectors to the actors modal when it gets opened from a trend

## Changes
- Returns all the selectors when needed back in the `to_actors_query_options` method
- Filters the underlying trends query when selecting the filters in the actors modal
  - This required a fair amount of wrangling in the `query_builder.py` class. I'm not so happy with this, as it adds further complexity, but, unfortunately, trends is such a complex query that this is necessary
- I've commented out the code for returning/filtering the results for when `compare` is turned on - more details below

<img width="570" alt="image" src="https://github.com/PostHog/posthog/assets/1459269/eba4c1fe-6b26-4af6-93ad-074c0eac0569">

## TODO:

I want to get to as much of this as I can, but ultimately I won't be able to:
- [x]  ~Finish off the `compare` mode stuff~ ✅ 
  - ~This is hard due to how `days` gets returned for the "previous" results. We return the previous intervals days, and so reading the `days` from the persons modal can return two different sets of data depending on which series you click on, and thus messing up the modals selectors~
- [ ]  Adding tests for the return of `to_actors_query` in the trends query runner. It _looks_ like its returning different data, but I haven't validated the changes I've made to the query builder yet, and so some deep testing here will be needed
- [ ]  There's a bug where opening the actors modal always defaults the day selector field to the first day of the series and not the day that was selected
- [x]  ~Query snapshots seems like some old tests are now running extra queries when they shouldn't be - looks like they're running extra breakdown values queries. I think this is down to how I changed the `Breakdown(..)` object parent method from a cached property to a normal function, and thus it looks like the underlying query may be getting ran on every func init, maybe~ ✅ 
- [ ]  Breakdowns on "Other" returns all data it seems, as opposed to being filtered on just the other values

## How did you test this code?
- Unit test and browser clicky clicks

Gilbert09 committed:
Added breakdowns and series selectors to the actors modal

github-actions[bot] committed:
Update query snapshots

mariusandra reviewed (approved):
Nice work. This does look like it goes deep. Since this touches almost only flagged code, I have nothing against getting it in, so here's a ✅. However the case with the double breakdown query sounds like something we'll miss if we don't follow up.

Once change I'd also make is to add the same empty array check to `InsightActorsQueryOptions`, to make the events/person explorer look better when there's no breakdown.

Gilbert09 committed:
Added compare to the actor modal

Gilbert09 committed:
Fixed breakdowns making a bazillon queries

github-actions[bot] committed:
Update query snapshots

Gilbert09 committed:
Merge branch 'master' into tom/trends-actor-query-breakdowns

webjunkie committed:
Merge branch 'master' into tom/trends-actor-query-breakdowns

webjunkie committed:
Repair modal state logic cause loadActors only uses actorsQuery selector

I suppose this was changed at some point. We cannot use the query
argument anymore.

webjunkie committed:
Adjust the breakdown limit when not hiding other data

Aligns with the legacy version

github-actions[bot] committed:
Update query snapshots

github-actions[bot] committed:
Update query snapshots

webjunkie committed:
Adjust tests

webjunkie committed:
Turn off "Other" breakdown option for modal for now

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

webjunkie committed:
Fix test

webjunkie committed:
Merge branch 'master' into tom/trends-actor-query-breakdowns

# Conflicts:
#	frontend/__snapshots__/scenes-app-insights-error-empty-states--long-loading--light.png

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

webjunkie committed:
Fix tests

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

github-actions[bot] committed:
Update query snapshots

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

webjunkie commented:
I merged this PR so it doesn't linger around longer.

From Tom's open todos:

> Adding tests for the return of `to_actors_query` in the trends query runner. It _looks_ like its returning different data, but I haven't validated the changes I've made to the query builder yet, and so some deep testing here will be needed

Still needs to be done.

> There's a bug where opening the actors modal always defaults the day selector field to the first day of the series and not the day that was selected

I could not replicate this.

> Breakdowns on "Other" returns all data it seems, as opposed to being filtered on just the other values

I didn't want to pour more time looking into this, not sure how to handle. To mitigate meanwhile, I removed the "other" options in the modals, so you can filter by what works.
